app/src/main/java/com/github/gogetters/letsgo/activities/ChatActivity.kt

```diff
@@ -1,151 +1,112 @@
 package com.github.gogetters.letsgo.activities
 
 import android.os.Bundle
-import android.util.Log
-import android.view.View
-import android.widget.EditText
-import android.widget.ListView
-import android.widget.Toast
+import android.provider.ContactsContract
 import androidx.appcompat.app.AppCompatActivity
 import com.github.gogetters.letsgo.R
-import com.github.gogetters.letsgo.chat.ChatMessage
-import com.github.gogetters.letsgo.chat.MessageAdapter
+import com.github.gogetters.letsgo.chat.model.ChatMessageData
+import com.github.gogetters.letsgo.chat.model.UserData
+import com.github.gogetters.letsgo.chat.views.ChatMyMessageItem
+import com.github.gogetters.letsgo.chat.views.ChatTheirMessageItem
 import com.github.gogetters.letsgo.database.Database
-import com.github.gogetters.letsgo.database.types.MessageData
-import com.google.firebase.auth.FirebaseAuth
 import com.google.firebase.database.ChildEventListener
-import com.google.firebase.database.ktx.getValue
-import java.util.*
-
+import com.google.firebase.database.DataSnapshot
+import com.google.firebase.database.DatabaseError
+import com.google.firebase.database.FirebaseDatabase
+import com.xwray.groupie.GroupAdapter
+import com.xwray.groupie.ViewHolder
+import kotlinx.android.synthetic.main.activity_chat.*
 
 class ChatActivity : AppCompatActivity() {
 
-    private val _defaultUsername = "Default User"
-    private val _testingChatId = "testingChatId"
-
-    // current user of the chatApp
-    private lateinit var userName: String
-    // current user id of the chatApp
-    private var userId: String? = null
-    // placeholder for chat bubbles
-    private lateinit var listView: ListView
-    // text message to send
-    private lateinit var entryText: EditText
-    // update connection to listview
-    private lateinit var adapter: MessageAdapter
-    private lateinit var messageListener: ChildEventListener
-
-    // connect to the Chat Collection's Message Document in Firestore
-//    private val firestoreChat by lazy {
-//        FirebaseFirestore.getInstance().collection(COLLECTION_KEY).document(DOCUMENT_KEY)
-//    }
-
-    // data structure in Firestore
+    val adapter = GroupAdapter<ViewHolder>()
+    var userId: String? = null
+    var toUser: UserData? = null
+
     companion object {
-        const val COLLECTION_KEY = "Chat"
-        const val DOCUMENT_KEY = "Message"
-        const val NAME_FIELD = "Name"
-        const val TEXT_FIELD = "Text"
+        val UNKNOWN = "Unknown"
```

Comment:
this as well, if it is used for the userId maybe it should have some reference to the ID in the name